From 05f123e03eceb80328e35495f623b1da45caeb16 Mon Sep 17 00:00:00 2001
From: Maurizio Lombardi <mlombard@redhat.com>
Date: Mon, 8 Sep 2014 11:07:44 -0400
Subject: [scsi] bnx2i, be2iscsi: fix custom stats length

Message-id: <1410174472-14981-10-git-send-email-mlombard@redhat.com>
Patchwork-id: 91642
O-Subject: [RHEL7.1 e-stor PATCH 09/17] bnx2i, be2iscsi: fix custom stats length
Bugzilla: 1089400
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Rob Evers <revers@redhat.com>

From: Mike Christie <michaelc@cs.wisc.edu>

The custom stats is an array with custom_length indicating the length
of the array. This patch fixes bnx2i and be2iscsi's setting of the
custom stats length. They both just have the one, eh_abort_cnt, so that should
be in the first entry of the custom array and custom_length should then
be one.

Reported-by: Rickard Strandqvist <rickard_strandqvist@spectrumdigital.se>
Signed-off-by: Mike Christie <michaelc@cs.wisc.edu>
Acked-by: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Acked-by: Eddie Wai <eddie.wai@broadcom.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 915aafd856d1a4ef1dea30c2b20ada03c93be4d7)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/be2iscsi/be_iscsi.c b/drivers/scsi/be2iscsi/be_iscsi.c
index 627d6e0..bc74f55 100644
--- a/drivers/scsi/be2iscsi/be_iscsi.c
+++ b/drivers/scsi/be2iscsi/be_iscsi.c
@@ -916,7 +916,7 @@ void beiscsi_conn_get_stats(struct iscsi_cls_conn *cls_conn,
  stats->r2t_pdus = conn->r2t_pdus_cnt;
  stats->digest_err = 0;
  stats->timeout_err = 0;
- stats->custom_length = 0;
+ stats->custom_length = 1;
  strcpy(stats->custom[0].desc, "eh_abort_cnt");
  stats->custom[0].value = conn->eh_abort_cnt;
 }
diff --git a/drivers/scsi/bnx2i/bnx2i_iscsi.c b/drivers/scsi/bnx2i/bnx2i_iscsi.c
index add22ae..8598b64 100644
--- a/drivers/scsi/bnx2i/bnx2i_iscsi.c
+++ b/drivers/scsi/bnx2i/bnx2i_iscsi.c
@@ -1643,12 +1643,11 @@ static void bnx2i_conn_get_stats(struct iscsi_cls_conn *cls_conn,
  stats->r2t_pdus = conn->r2t_pdus_cnt;
  stats->tmfcmd_pdus = conn->tmfcmd_pdus_cnt;
  stats->tmfrsp_pdus = conn->tmfrsp_pdus_cnt;
- stats->custom_length = 3;
- strcpy(stats->custom[2].desc, "eh_abort_cnt");
- stats->custom[2].value = conn->eh_abort_cnt;
  stats->digest_err = 0;
  stats->timeout_err = 0;
- stats->custom_length = 0;
+ strcpy(stats->custom[0].desc, "eh_abort_cnt");
+ stats->custom[0].value = conn->eh_abort_cnt;
+ stats->custom_length = 1;
 }
 
 
-- 
1.7.1